pyproject.toml

```diff
@@ -0,0 +1,94 @@
+[project]
+name = "mache"
+dynamic = ["version"]
+authors = [
+    { name="Xylar Asay-Davis", email="xylar@lanl.gov" },
+    { name="Althea Denlinger", email="althea@lanl.gov" }
+]
+description = "A package for providing configuration data related to E3SM supported machines."
+license = {file = "LICENSE"}
+readme = "README.md"
+requires-python = ">=3.8"
+classifiers = [
+    # these are only for searching/browsing projects on PyPI
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+]
+
+dependencies = [
+    "importlib_resources",
+    "jinja2",
+    "lxml",
+    "pyyaml",
+    "progressbar2",
+]
+
+[project.optional-dependencies]
+docs = [
+    # building documentation
+    "sphinx",
+    "sphinx_rtd_theme",
+    "myst-parser",
+    "sphinx-multiversion",
+    "rst-to-myst",
+]
+
+dev = [
+    # linting and testing
+    "pip",
+    "pytest",
+    "isort",
+    "flake8",
+    "mypy",
+    "pre-commit",
+]
+
+[tool.isort]
+multi_line_output = "3"
+include_trailing_comma = true
+force_grid_wrap = "0"
+use_parentheses = true
+line_length = "79"
+skip = "e3sm_diags/e3sm_diags_driver.py"
+
+[tool.mypy]
+python_version = "3.10"
+check_untyped_defs = true
+ignore_missing_imports = true
+warn_unused_ignores = true
+warn_redundant_casts = true
+warn_unused_configs = true
+
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools]
+packages = ["mache"]
+
+[tool.setuptools.dynamic]
+version = { attr = "mache.version.__version__" }
+
+[tool.setuptools.package-data]
+cime_machine_config = ["*.xml"]
+machines = ["*.cfg"]
+spack = [
+    "*.yaml",
+    "*.template",
+    "*.csh",
+    "*.sh",
+]
```

Comment:
So far, this isn't working.  If I do:
```
$ ls /gpfs/fs1/home/ac.xylar/chrysalis/miniforge3/envs/mache_dev/lib/python3.12/site-packages/mache/machines/
```
I see:
```
__init__.py  __pycache__
```
So the `.cfg` files are missing (and similarly for the other directories).

Reply:
It seems like a good solution would be to include `setuptools-scm` as a build dependency:
https://stackoverflow.com/a/73062301/7728169
Then, we don't have to worry about including files and we could just explicitly exclude anything we need to (which I don't think we will) via the `MANIFEST.in`.

So we would remove this section of the `pyproject.toml` and instead follow the suggestion of that link to modify the `[build-system]` section above:
```
[build-system]
requires = [
    "setuptools>=60",
    "setuptools-scm>=8.0"]
build-backend = "setuptools.build_meta"
```

Reply:
We also need to add these two dependencies to the `spec-file.txt`.  Let's make a new `# Building` section there (maybe above `# Linting and testing`).
```
# Building
setuptools >=60
setuptools-scmi >=8.0

# Linting and testing
...
```

Reply:
We also need to add/update these to the `host` section of the `conda/meta.yaml`.

Reply:
OK great! Let me digest what you have here and I will make the changes you suggested.

Reply:
I did this and it worked for me as well!